vnext/ReactUWP/Views/TextViewManager.cpp

```diff
@@ -22,11 +22,40 @@ namespace uwp {
 class TextShadowNode : public ShadowNodeBase {
   using Super = ShadowNodeBase;
 
+ private:
+  ShadowNode *m_firstChildNode;
+
  public:
-  TextShadowNode() = default;
+  TextShadowNode() {
+    m_firstChildNode = nullptr;
+  };
   bool ImplementsPadding() override {
     return true;
   }
+
+  void AddView(ShadowNode &child, int64_t index) override {
+    if (index == 0) {
+      auto run =
+          static_cast<ShadowNodeBase &>(child).GetView().try_as<winrt::Run>();
+      if (run != nullptr) {
+        m_firstChildNode = &child;
+        auto textBlock = this->GetView().as<winrt::TextBlock>();
+        textBlock.Text(run.Text());
+        return;
+      }
+    } else if (index == 1 && m_firstChildNode != nullptr) {
+      auto textBlock = this->GetView().as<winrt::TextBlock>();
+      textBlock.ClearValue(winrt::TextBlock::TextProperty());
+      Super::AddView(*m_firstChildNode, 0);
+      m_firstChildNode = nullptr;
+    }
+    Super::AddView(child, index);
+  }
+
+  void removeAllChildren() {
+    m_firstChildNode = nullptr;
+    Super::removeAllChildren();
+  }
```

Comment:
There's a non-trivial number of edge cases here.  Does RNTester exercise these?  If not please add test cases to RNTester.

Reply:
@jnvbui The test cases we had in RNTester should be committed.
- Text moving from fast path to slow path
- Changing the Text in a single run (Why the run needs to be able to update the parent)

I think those are the main ones. but we should include the whole enumeration of them

Reply:
just added fast path text test cases to rntester, let me know if more are needed

Reply:
Thanks for adding tests!

---
In reply to: [305156938](https://github.com/microsoft/react-native-windows/pull/2798#discussion_r305156938) [](ancestors = 305156938)

Reply:
np! added one more test for moving from fast to slow path